libp2p/Common.h

```diff
@@ -172,6 +172,7 @@ struct NodeIPEndpoint
 	/// Setting true causes isAllowed to return true for all addresses. (Used by test fixtures)
 	static bool test_allowLocal;
 
+	NodeIPEndpoint(): address(bi::address()), udpPort(0), tcpPort(0) {}
```

Comment:
My advice would be to use non-static member initialization feature: `uint16_t udpPort = 0` instead of initializing in to 0 in many constructors. Same can be done for `address`.